activesupport/test/broadcast_logger_test.rb

```diff
@@ -290,6 +290,17 @@ def info(msg, &block)
       assert(logger.qux(param: "foo"))
     end
 
+    test "#deep_dup duplicates the broadcasts" do
+      logger = CustomLogger.new
+      logger.level = ::Logger::WARN
+      broadcast_logger = BroadcastLogger.new(logger)
+
+      duplicate = broadcast_logger.deep_dup
+
+      assert_equal ::Logger::WARN, duplicate.broadcasts.sole.level
```

Comment:
Considering my other comments, I'd suggest adding assertions that confirm the original `BroadcastLogger` (and its list of `@broadcasts`) was not altered, and is not tied to its duplicate.
```rb
assert_equal [logger], broadcast_logger.broadcasts
```

Reply:
I just tried it and got this error:
```
Failure:
ActiveSupport::BroadcastLoggerTest#test_#deep_dup_duplicates_the_broadcasts [activesupport/test/broadcast_logger_test.rb:300]:
No visible difference in the ActiveSupport::BroadcastLoggerTest::CustomLogger#inspect output.
You should look at the implementation of #== on ActiveSupport::BroadcastLoggerTest::CustomLogger or its members.
#<ActiveSupport::BroadcastLoggerTest::CustomLogger:0xXXXXXX @adds=[], @closed=false, @chevrons=[], @level=2, @local_level=0, @progname=nil, @formatter=nil>
```
Very curious. Looking at the [doc for MiniTest's `assert_equal`](https://docs.ruby-lang.org/en/2.1.0/MiniTest/Assertions.html#method-i-assert_equal), I found this:

> If there is no visible difference but the assertion fails, you should suspect that your #== is buggy, or your inspect output is missing crucial details.

Instead I tried using `assert_same`:
```rb
      assert_same logger, broadcast_logger.broadcasts.sole
      refute_same logger, duplicate.broadcasts.sole
      assert_equal ::Logger::WARN, duplicate.broadcasts.sole.level
```

And then I get the failure I expected:
```
Failure:
ActiveSupport::BroadcastLoggerTest#test_#deep_dup_duplicates_the_broadcasts [activesupport/test/broadcast_logger_test.rb:300]:
Expected #<ActiveSupport::BroadcastLoggerTest::CustomLogger:0x00000001030de158 @adds=[], @closed=false, @chevrons=[], @level=2, @local_level=0, @progname=nil, @formatter=nil> (oid=5640) to be the same as #<ActiveSupport::BroadcastLoggerTest::CustomLogger:0x00000001030de1f8 @adds=[], @closed=false, @chevrons=[], @level=2, @local_level=0, @progname=nil, @formatter=nil> (oid=5660).
```